File: tests/cypress/cypress/elements/pages/files/upload_file.rb
class UploadFile < FileManagerPage
  set_url_matcher /files\/upload/

  # Main box elements
  element :heading, 'div.form-standard form div.form-btns-top h1'

  # Edit form
  element :file_input, 'div.col.w-12 div.form-standard form fieldset input[name="file"]'
  element :title_input, 'div.col.w-12 div.form-standard form fieldset input[name="title"]'
  element :description_input, 'div.col.w-12 div.form-standard form fieldset textarea[name="description"]'
  element :credit_input, 'div.col.w-12 div.form-standard form fieldset input[name="credit"]'
  element :location_input, 'div.col.w-12 div.form-standard form fieldset input[name="location"]'
  element :form_submit_button, 'div.form-standard form div.form-btns-top input[type="submit"]'

  def load
    click_link 'Files'
    click_link 'Upload File'
    within '.section-header__controls .filter-submenu' do
      click_link 'Main Upload Directory'
    end
  end

end

File: tests/cypress/cypress/elements/pages/files/UploadFile.js
import FileManagerSection from '../_sections/FileManagerSection'
const { $ } = Cypress
class UploadFile extends FileManagerSection {
  constructor() {
      super()
      this.urlMatch = /files\/upload/;

      this.elements({
        // Main box elements
        'heading': 'div.form-standard form .form-btns h3',

        // Edit form
        'file_input': 'div[data-input-value="files_field"]',
        'title_input': '.ee-main__content div.form-standard form input[type!=hidden][name="title"]',
        'description_input': '.ee-main__content div.form-standard form textarea[name="description"]',
        'credit_input': '.ee-main__content div.form-standard form input[type!=hidden][name="credit"]',
        'location_input': '.ee-main__content div.form-standard form input[type!=hidden][name="location"]',
        'form_submit_button': '.ee-main__content div.form-standard form .form-btns-top [type="submit"]'

      })
    }
    load() {
      cy.visit('admin.php?/cp/files')
      cy.get('.sidebar').contains('Main Upload Directory').click()
    }

    dragAndDropUpload(file) {
        cy.get('.file-upload-widget').then(function(widget) {
            $(widget).removeClass('hidden')
        })
        cy.intercept('/admin.php?/cp/addons/settings/filepicker/ajax-upload').as('upload')
        cy.intercept('/admin.php?/cp/files/directory/*').as('table')
        this.get('file_input').find('.file-field__dropzone').attachFile(file, { subjectType: 'drag-n-drop' })
        cy.wait('@upload')
    }

}
export default UploadFile;
